src/components/pages/ImageClassification/ImageUploadModal/ImageUploadModal.js

```diff
@@ -55,82 +59,72 @@ const ImageUploadModal = ({ isOpen, onClose, onFilesUpload, existingFiles }) =>
     })
   }
 
+  const processSingleImage = async (file) => {
+    try {
+      const imageData = await databaseSwitchboardInstance.uploadImage(projectId, recordId, file)
+
+      // Placeholder: Polling logic will go here.
+      // TODO: Implement polling to check the status of image processing before proceeding to the next image.
+      // something like: `await pollForCompletion(imageData.id);`
+
+      setProcessedFiles((prev) => prev + 1)
+      return imageData
+    } catch (error) {
+      toast.error(`Failed to upload ${file.name}: ${error.message}`)
+      return null
+    }
+  }
+
   const validateAndUploadFiles = async (files) => {
     setLoading(true)
     setTotalFiles(files.length)
     setProcessedFiles(0)
     isCancelledRef.current = false
 
-    const validFiles = []
-    const invalidFiles = []
-    const duplicateFiles = []
-    const oversizedFiles = []
-    const dimensionExceededFiles = []
-    const corruptFiles = []
-
-    const allFiles = [...existingFiles]
+    const uploadedFiles = []
 
-    for (const [index, file] of files.entries()) {
+    for (const file of files) {
       if (isCancelledRef.current) {
         setLoading(false)
         return
       }
 
+      // Validate file type, size, dimensions, and uniqueness.
       if (!validFileTypes.includes(file.type)) {
-        invalidFiles.push(file)
-      } else if (file.size > maxFileSize) {
-        oversizedFiles.push(file)
-      } else if (allFiles.some((existingFile) => existingFile.name === file.name)) {
-        duplicateFiles.push(file)
-      } else {
-        const result = await validateDimensions(file)
-        if (isCancelledRef.current || result.cancelled) {
-          setLoading(false)
-          return
-        }
-
-        if (result.valid && !result.corrupt) {
-          validFiles.push(result.file)
-        } else if (result.corrupt) {
-          corruptFiles.push(result.file)
-        } else {
-          dimensionExceededFiles.push(result.file)
-        }
+        toast.error(`Invalid file type: ${file.name}`)
+        continue
```

Comment:
added `continue` statements in the if blocks to skip over invalid files without halting the entire process